src/psyclone/tests/domain/lfric/lfric_builtins_test.py

```diff
@@ -688,3444 +691,903 @@ def test_X_plus_Y(
-            "      CALL f2_proxy%set_dirty()\n"
-            "      !\n")
-        if not annexed:
-            output_dm_2 = output_dm_2.replace("dof_annexed", "dof_owned")
-        assert output_dm_2 in code
+            "f2_data(df) = f1_data(df)\n") in code
 
 
-# ------------- Sign of real field elements --------------------------------- #
+def test_setval_random(fortran_writer):
+    ''' Test the metadata, str and lower_to_language_level builtin methods. '''
+    metadata = lfric_builtins.LFRicSetvalRandomKern.metadata()
+    assert isinstance(metadata, LFRicKernelMetadata)
+    assert len(metadata.meta_args) == 1
+    assert metadata.meta_args[0].access == "gh_write"
+    assert metadata.meta_args[0].function_space == "any_space_1"
 
+    kern = builtin_from_file("15.7.4_setval_random_builtin.f90")
+    assert str(kern) == ("Built-in: setval_random (fill a real-valued field"
+                         " with pseudo-random numbers)")
 
-def test_sign_X(tmpdir, monkeypatch, annexed, dist_mem, fortran_writer):
-    '''Test that 1) the '__str__' method of 'LFRicSignXKern' returns the
-    expected string and 2) we generate correct code for the built-in
-    operation 'Y = sign(a, X)' where 'a' is a real scalar and 'Y' and
-    'X' are real-valued fields. Test with and without annexed DoFs
-    being computed as this affects the generated code. 3) Also test the
-    'lower_to_language_level()' and 'metadata()' methods.
+    # Test the 'lower_to_language_level()' method
+    code = fortran_writer(kern)
+    assert ("! Built-in: setval_random (fill a real-valued field "
+            "with pseudo-random numbers)\n"
+            "call RANDOM_NUMBER(f1_data(df))\n") in code
 
-    '''
+
+# ------------- Sign of real field elements --------------------------------- #
+
+
+def test_sign_X(fortran_writer):
+    ''' Test the metadata, str and lower_to_language_level builtin methods. '''
     metadata = lfric_builtins.LFRicSignXKern.metadata()
     assert isinstance(metadata, LFRicKernelMetadata)
-    api_config = Config.get().api_conf(API)
-    monkeypatch.setattr(api_config, "_compute_annexed_dofs", annexed)
-    _, invoke_info = parse(os.path.join(BASE_PATH,
-                                        "15.10.1_sign_X_builtin.f90"),
-                           api=API)
-    psy = PSyFactory(API, distributed_memory=dist_mem).create(invoke_info)
-    # Test '__str__' method
-    first_invoke = psy.invokes.invoke_list[0]
-    kern = first_invoke.schedule.children[0].loop_body[0]
+    assert len(metadata.meta_args) == 3
+    assert metadata.meta_args[0].access == "gh_write"
+    assert metadata.meta_args[0].function_space == "any_space_1"
+    assert metadata.meta_args[1].access == "gh_read"
+    assert metadata.meta_args[2].access == "gh_read"
+    assert metadata.meta_args[2].function_space == "any_space_1"
+
+    kern = builtin_from_file("15.10.1_sign_X_builtin.f90")
     assert str(kern) == "Built-in: sign_X (sign of a real-valued field)"
```

Comment:
Not yours but "...field, applied to a scalar argument)" perhaps as it took me a while to work out what it does.

Reply:
Done